action.yml

```diff
@@ -86,10 +86,26 @@ inputs:
       'A list of environment variable(s) for the container. Space-separated values in 'key=value' format. Empty string
       to clear existing values. Prefix value with 'secretref:' to reference a secret.'
     required: false
+  disableTelemetry:
+    description: |
+      'If set to "true", no telemetry will be collected by this GitHub Action. If set to "false", or if this argument is
+      not provided, telemetry will be sent to Microsoft about the Container App build and deploy scenario targeted by
+      this GitHub Action.'
+    required: false
+    default: false
 
 runs:
   using: "composite"
   steps:
+    - name: Start timer to track action execution length, in milliseconds
+      if: ${{ always() && inputs.disableTelemetry == 'false' }}
```

Comment:
OK I'm not an expert in Github Action, but is this `always()` something we should remove from the `if`? Or is there are legitimate reason to keep it (in which case the rest of the condition doesn't seem useful?)?

Reply:
Similar question for the other always() below

Reply:
From the [GitHub documentation](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-not-requiring-successful-dependent-jobs), `always()` is run independent of the preceding steps results, _i.e._, if any step fails prior to the step which includes `always()` in the conditional, the step will still run (since workflows will terminate when the action fails).

I'm using the `always()` conditional in a few places here to ensure that even if the action fails, we can still record telemetry about the run, specifically how long it executed for and that the result was `failed` (this may be able to help us catch timeouts/long-running actions, for example). I thought about adding an additional property for failed runs for which step it failed on, but decided against it since it would be a bit more involved and we can decide later if that's a piece of information we're interested in collecting.